File: server/src/main/java/com/yawa/server/components/environment/AppEnvironment.java
package com.yawa.server.components.environment;

import lombok.Data;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
@Data
public class AppEnvironment {

    @Value("${yawa.stack}")
    private String stack;

    @Value("${yawa.region}")
    private String region;

    @Value("${yawa.application}")
    private String application;

    @Value("${yawa.service}")
    private String service;
}

File: server/src/main/kotlin/com/yawa/server/components/environment/AppEnvironment.kt
package com.yawa.server.components.environment

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class AppEnvironment(

    @Value("\${yawa.stack}")
    val stack: String,

    @Value("\${yawa.region}")
    val region: String,

    @Value("\${yawa.application}")
    val application: String,

    @Value("\${yawa.service}")
    val service: String
)
